src/components/Preferences/ScheduleSelector.tsx

```diff
@@ -1,53 +1,61 @@
-import React from "react";
+import React, { useState } from "react";
 import ScheduleSelector from "react-schedule-selector";
 
-type PropsType = { color: string; semester: string };
-class Timetable extends React.Component<PropsType> {
-    state = { schedule: [], data: [], semester: this.props.semester };
-    today = new Date();
-    first = this.today.getDate() - this.today.getDay() + 1;
-    unselectedColorDict = {
-        Blue: "rgba(162, 198, 248, 1)",
-        Orange: "rgba(255, 165, 0)",
-        Pink: "rgba(255, 182, 193)",
+const Timetable = (props) => {
+    const { semester, values, setFieldValue } = props;
+    const today = new Date();
+    const first = today.getDate() - today.getDay() + 1;
+    const form_value = "preferredTime." + semester;
+    const datetime_arr = [];
```

Comment:
sorry, I missed these before, should be camelcase
looks good otherwise

Reply:
Oh sorry! I merged this. Will solve this with my next PR